File: src/main/java/com/kid/A0/service/Interface/PlanServiceInterface.java
package com.kid.A0.service.Interface;

import com.kid.A0.dto.PlanResponse;
import com.kid.A0.dto.PlanUpdate;

import java.util.List;

public interface PlanServiceInterface {

    List<PlanResponse> getPlans();

    PlanResponse createPlan(PlanUpdate planUpdate);

    PlanResponse getPlan(long id);

    void deletePlan(long id);

    PlanResponse updatePlan(long id, PlanUpdate planUpdate);

    PlanResponse replacePlan(long id, PlanUpdate planUpdate);
}

File: src/main/java/com/kid/A0/service/Interface/PlanServiceInterface.kt
package com.kid.A0.service.Interface

import com.kid.A0.dto.PlanResponse
import com.kid.A0.dto.PlanUpdate

interface PlanServiceInterface {

    val plans: List<PlanResponse>

    fun createPlan(planUpdate: PlanUpdate): PlanResponse

    fun getPlan(id: Long): PlanResponse

    fun deletePlan(id: Long)

    fun updatePlan(id: Long, planUpdate: PlanUpdate): PlanResponse

    fun replacePlan(id: Long, planUpdate: PlanUpdate): PlanResponse
}